### ISSUE TITLE
Loading application language strings outside of package namespace

### ISSUE THREAD

**Issue by kohsah** [cirsfid-unibo/lime#34]
The following functionality was removed in d97411fa3ed586423c25f5ecb19ed7a419bfa0ab [1]

I think this functionality was actually very useful to have. For example, if you have to change an existing string used in lime-core you would have to modify the package or override it from another package.  With this config/local/lang-xy.json file we could override the package string since it is loaded last... so we can change strings without modifying lime-core or having a package just to override strings. I suggest that having this ability to load config/locale/lang-xy.json should be brought back. 

[1]

```
loadLanguage: function() {
        var langUrl = 'config/locale/lang-'+this.config.lang+'.json';
        var extLangUrl = 'config/locale/ext/ext-lang-'+this.config.lang+'.js';
        Ext.Ajax.request({
            url : langUrl,
            async: false,
            scope: this,
            success : function(response, opts) {
                try {
                    this.strings = Ext.decode(response.responseText);
                    this.isReady = true;
                } catch(e) {
                    alert("Fatal error on loading localization files");
                }
            },
            failure : function(response, opts) {
                alert("Fatal error on loading localization files");
            }
        });

        Ext.Loader.loadScript({url : extLangUrl});
    }

```

**Comment by obujor** [comment]
Yes, this functionality was removed because of performace reasons in production mode, we're trying to reduce the amount of asynchronous requests and load the app as soon as possible.

I understand what you're talking about, now is more difficult to add a string to lime-core but there is a reason and it is the separation of strings by packages, below there is an explanation.

The strings files in 'config/locale' were added when the packages did not exist. Now that we have packages we want to put inside every package the used strings and in this allows us to compile them into the application without the need to download all the strings files on loading.
So now the strings are in following files:
packages/lime-core/str/Strings.js -> In this files are contained all the 'config/locale' strings because the lime-core components are using these strings so it's important that this file is containing only the strings used by lime-core.
packages/somePackage/str/Strings.js -> Contains the strings used by 'somePackage' only.
languagesPlugins/akoma3.0/interface/default/LANG/markupMenu.json -> Contains the translation of the buttons strings.

So for now all the strings that LIME needs are included in all these files and I think there is not reason to return to old config/locale, if you think about it the strings are quite static if you need to add a new string this means that new feature has added which uses this string so you need to build the project anyway. If you want to change some strings there is no problem to modify the Strings.js of the package and to build LIME. 
Is difficult to find a scenario when you need to change a string directly in production mode.

**Comment by kohsah** [comment]
OK, I get that JS is more performant than asynchronously loading JSON.  I don't see reason to add new strings in lime-core, but there can be cases where you have to override a string (lets say i want to show a different error message than the standard one... the way then would be to override lime-core and overwrite that particular string in the override package ?

**Comment by obujor** [comment]
Yes it could happen that you want to change a string because of your preference, in my opinion the best way to do such things is to create a package containing all the changes (strings and/or code) that you want to change for let say a specific customer.

For example you can create a package called "lime-kohsah" which overrides some files:
packages/lime-core/src/Strings.js -> in this you're going to change the strings that you don't like in the core
packages/akn-metadata/src/Strings.js -> in this you can change the metadata editor strings

This gives you the advantage to change the customization just by including or removing some packages from app.json, so if you need for some reasons to have different set of translations because of the preference of your customers you can simply enable his specific package and build the app.

If you're looking to change only the strings this approach seems to cause some structure overhead, but if you think that with a custom package you can change almost everything, including the sass style and sass variables it becomes a powerful approach to customize LIME.

**Comment by kohsah** [comment]
Indeed i have a `fao-lime` package with overrides, will use that method.
